frontend/src/tests/components/OurTable.test.js

```diff
@@ -1,674 +1,432 @@
+    ).toBeInTheDocument();
+    expect(
+      await screen.findByTestId("testid-current-page-button")
+    ).toBeInTheDocument();
+  });
+
+  test("renders a table with 11 rows and clicks on the next page button and previous page button", async () => {
+    render(<OurTable columns={columns} data={elevenRows} />);
+
+    expect(
+      await screen.findByTestId("testid-next-page-button")
+    ).toBeInTheDocument();
+    expect(
+      (await screen.findByTestId("testid-prev-page-button")).hasAttribute(
+        "disabled"
+      )
+    ).toBe(true);
+    expect(
+      (await screen.findByTestId("testid-next-page-button")).hasAttribute(
+        "disabled"
+      )
+    ).toBe(false);
+    const nextButton = screen.getByTestId("testid-next-page-button");
+    expect(await screen.findByText(`Hello 10`)).toBeInTheDocument();
+    expect(screen.queryByText(`Hello 11`)).not.toBeInTheDocument();
+    expect(
+      await screen.findByTestId("testid-current-page-button")
+    ).toContainHTML("1");
+    fireEvent.click(nextButton);
+    expect(
+      await screen.findByTestId("testid-current-page-button")
+    ).toContainHTML("2");
+    expect(
+      (await screen.findByTestId("testid-prev-page-button")).hasAttribute(
+        "disabled"
+      )
+    ).toBe(false);
+    expect(
+      (await screen.findByTestId("testid-next-page-button")).hasAttribute(
+        "disabled"
+      )
+    ).toBe(true);
+    expect(await screen.findByText(`Hello 11`)).toBeInTheDocument();
+    const prevButton = screen.getByTestId("testid-prev-page-button");
+    fireEvent.click(prevButton);
+    expect(await screen.findByText(`Hello 1`)).toBeInTheDocument();
+  });
+
+  test("renders a table with 51 rows and tests the first page", async () => {
+    render(<OurTable columns={columns} data={fiftyOneRows} />);
+    expect(
+      await screen.findByTestId("testid-next-page-button")
+    ).toBeInTheDocument();
+    expect(screen.queryByTestId("testid-left-ellipsis")).toBe(null);
+    expect(screen.queryByTestId("testid-back-three-page-button")).toBe(null);
+    expect(screen.queryByTestId("testid-back-two-page-button")).toBe(null);
+    expect(screen.queryByTestId("testid-back-one-page-button")).toBe(null);
+    expect(
+      await screen.findByTestId("testid-current-page-button")
+    ).toContainHTML("1");
+    expect(
+      await screen.findByTestId("testid-forward-one-page-button")
+    ).toContainHTML("2");
+    expect(
+      await screen.findByTestId("testid-forward-two-page-button")
+    ).toContainHTML("3");
+    expect(screen.queryByTestId("testid-forward-three-page-button")).toBe(null);
+    expect(
+      await screen.findByTestId("testid-right-ellipsis")
+    ).toBeInTheDocument();
+    expect(await screen.findByTestId("testid-last-page-button")).toContainHTML(
+      "6"
+    );
+    expect(await screen.findByText(`Hello 10`)).toBeInTheDocument();
+  });
+
+  test("renders a table with 51 rows and tests moving forward and back one page", async () => {
```

Comment:
I think this is a strict subset of what the "move forward and back two pages tests" or at least they could be merged since they test very similar things

Reply:
This test also tests that the single page navigation buttons disable and enable when they're supposed to. I felt it relevant to keep these tests separate as opposed to making a single "mega test".